Content.Server/Chat/ChatConditions/HasComponentEntityChatCondition.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Linq;
+using Content.Server.Power.Components;
+using Content.Server.Radio.Components;
+using Content.Server.Station.Components;
+using Content.Server.Station.Systems;
+using Content.Shared.Bed.Sleep;
+using Content.Shared.Chat;
+using Content.Shared.Chat.Prototypes;
+using Content.Shared.Mobs.Systems;
+using Content.Shared.Radio;
+using Content.Shared.Radio.Components;
+using Robust.Shared.Network;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Chat.ChatConditions;
+
+/// <summary>
+/// Checks if the consumer is on a map with an active server, if necessary.
+/// </summary>
+[DataDefinition]
+public sealed partial class HasComponentEntityChatCondition : EntityChatCondition
+{
+
+    [Dependency] private readonly IEntityManager _entityManager = default!;
+    [Dependency] private readonly IComponentFactory _componentFactory = default!;
+
+    [DataField]
+    public string? Component;
+
+    public override HashSet<EntityUid> FilterConsumers(HashSet<EntityUid> consumers, Dictionary<Enum, object> channelParameters)
+    {
+        IoCManager.InjectDependencies(this);
+
+        var returnConsumers = new HashSet<EntityUid>();
+
+        if (Component != null)
```

Comment:
here  i see that early return would benefit readability - its called 'reduce nesting' :3